File: packages/react-native-autoplay/ios/hybrid/HybridInformationTemplate.swift
//
//  HybridInformationTemplate.swift
//  Pods
//
//  Created by Samuel Brucksch on 05.11.25.
//

import NitroModules

class HybridInformationTemplate: HybridInformationTemplateSpec {

    func createInformationTemplate(config: InformationTemplateConfig) throws {
        let template = InformationTemplate(config: config)
        TemplateStore.addTemplate(
            template: template,
            templateId: config.id
        )
    }

    func updateInformationTemplateSections(
        templateId: String,
        section: NitroSection
    ) throws -> Promise<Void> {
        return Promise.async {
            guard
                let template = TemplateStore.getTemplate(templateId: templateId)
                    as? InformationTemplate
            else {
                throw AutoPlayError.invalidTemplateType(
                    "\(templateId) is not an InformationTemplate"
                )
            }

            await template.updateSection(section: section)
        }
    }
}

File: packages/react-native-autoplay/android/src/main/java/com/margelo/nitro/swe/iternio/reactnativeautoplay/HybridInformationTemplate.kt
package com.margelo.nitro.swe.iternio.reactnativeautoplay

import com.margelo.nitro.core.Promise
import com.margelo.nitro.swe.iternio.reactnativeautoplay.template.AndroidAutoTemplate
import com.margelo.nitro.swe.iternio.reactnativeautoplay.template.InformationTemplate

class HybridInformationTemplate : HybridInformationTemplateSpec() {
    override fun createInformationTemplate(config: InformationTemplateConfig) {
        val context = AndroidAutoSession.getRootContext()
            ?: throw IllegalArgumentException("createInformationTemplate failed, carContext not found")

        val template = InformationTemplate(context, config)
        AndroidAutoTemplate.setTemplate(config.id, template)
    }

    override fun updateInformationTemplateSections(
        templateId: String, section: NitroSection
    ): Promise<Unit> {
        return Promise.async {
            val template = AndroidAutoTemplate.getTemplate<InformationTemplate>(templateId)
            template.updateSection(section)
        }
    }
}